### DISCUSSION TITLE
Publish & Verify `SimpleStorage` contract on Sepolia testnet on `Etherscan`

### DISCUSSION THREAD

**Discussion by pr0x2b** [Cyfrin/foundry-full-course-cu#1911] [Show and tell]
- `Course name`: Solidity Developer/ Smart contract devlopment with Solidity
- `Lesson 11`: Deploying your first smart contract

## Context
After deploying the smart-contract on sepolia testnet through `WalletConnect` (for MetaMask), I think it's cool to verify and publish the smart-contract on etherscan. This will allow us to view read and write contract on etherscan + we can interact directly with contract functions on etherscan. 

## Steps

1. Under the contract address page, next to the “Token Transfers” tab, you will be able to find the “Contract” tab. Then click on “Verify And Publish.”
2. Fill-up required details and click verify and publish.
3. Congrats! your smart-contract is now verified and you'll see green check saying `Exact match`

![image](https://github.com/Cyfrin/foundry-full-course-cu/assets/22565727/3eeb985a-0c06-4463-bc4f-869baf7ac4dc)

Here's my published contract for reference: [0xcab97d92b8da27b5a92acf906b4092368adb058e](https://sepolia.etherscan.io/address/0xcab97d92b8da27b5a92acf906b4092368adb058e#code)

![source-code-verification](https://github.com/Cyfrin/foundry-full-course-cu/assets/22565727/cdd6ebd2-5f07-4a06-9e95-cd38fc3f84a8)

**Comment by EngrPips** [comment]
Thanks for sharing this with us, Buddy.
